library/std/src/io/mod.rs

```diff
@@ -2408,6 +2445,34 @@ impl<T: BufRead, U: BufRead> BufRead for Chain<T, U> {
     fn consume(&mut self, amt: usize) {
         if !self.done_first { self.first.consume(amt) } else { self.second.consume(amt) }
     }
+
+    fn read_until(&mut self, byte: u8, buf: &mut Vec<u8>) -> Result<usize> {
+        let mut read = 0;
+        if !self.done_first {
+            read += self.first.read_until(byte, buf)?;
+
+            match buf.last() {
+                Some(b) if *b == byte => return Ok(read),
+                _ => self.done_first = true,
+            }
+        }
+        read += self.second.read_until(byte, buf)?;
+        Ok(read)
+    }
+
+    fn read_line(&mut self, buf: &mut String) -> Result<usize> {
+        let mut read = 0;
+        if !self.done_first {
+            read += self.first.read_line(buf)?;
+
+            match buf.as_bytes().last() {
+                Some(b) if *b == b'\n' => return Ok(read),
```

Comment:
Same issue here: If `first.read_line()` reports EOF (`Ok(0)`), then this will check if the pre-existing data in `buf` ends in a newline.

Can you add a test case for this?

Reply:
Done !